docs/lua-modules/mcp23017.md

```diff
@@ -0,0 +1,234 @@
+# Lua MCP23017 Module for NodeMCU / ESP8266
+
+| Since  | Origin / Contributor  | Maintainer  | Source  |
+| :----- | :-------------------- | :---------- | :------ |
+| 2020-04-10 | [Marcel P.](https://github.com/plomi-net) | [Marcel P.](https://github.com/plomi-net) | [mcp23017.lua](../../lua_modules/mcp23017/mcp23017.lua) |
+
+
+This Lua module provides access to the MCP23017 module.
+
+The [MCP23017](http://ww1.microchip.com/downloads/en/devicedoc/20001952c.pdf) is a port expander and provides 16 channels for inputs and outputs. 
+Up to 8 devices (128 channels) are possible by the configurable address (A0 - A2).
+
+Due to the 16 channels, 2 bytes are required for switching outputs or reading input signals. These are A and B. 
+A single pin can be set or a whole byte. 
+
+The numbering of the individual pins starts at 0 and ends with 15.
+The numbers up to 7 are on register GPIO A and from 8 to 15 on GPIO B.
+
+
+!!! important
+	The module requires `i2c` and `bit` C module built into firmware.
+
+
+### Require
+```lua
+mcp23017 = require("mcp23017")
+```
+
+## Example Script
+The example script can be found [here](../../lua_examples/mcp23017/mcp23017_example.lua)
+
+## setup()
+Configures the interface and sets the i2c pins and the address of the device.
+Automatically resets the device state (see `mcp23017.reset()`)
+
+#### Syntax
+`mcp23017.setup(address, customSCL, customSDA)`
```

Comment:
I am uncomfortable with I2C device drivers that want to set up the bus, since the bus is shared by potentially multiple devices.  Perhaps take the `i2c` bus id rather than `SCL` and `SDA` and put the `i2c.setup()` call in the example.  Taking the bus id is a good idea because we might support multiple I2C busses, too.

Reply:
Thanks for your comment and i agree.
I have now solved it to create an i2c setup before and submit the i2c id as parameter. 
Alternatively you can let the module do a setup for you.